hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/writebehind/WriteBehindEntryStoreQueueReplicationTest.java

```diff
@@ -26,50 +27,57 @@
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.TestHazelcastInstanceFactory;
 import com.hazelcast.test.annotation.ParallelJVMTest;
-import com.hazelcast.test.annotation.SlowTest;
+import com.hazelcast.test.annotation.QuickTest;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import java.util.concurrent.TimeUnit;
 
-import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 
 @RunWith(HazelcastParallelClassRunner.class)
-@Category({SlowTest.class, ParallelJVMTest.class})
+@Category({QuickTest.class, ParallelJVMTest.class})
 public class WriteBehindEntryStoreQueueReplicationTest extends HazelcastTestSupport {
 
     @Test
-    public void testWriteBehindQueueIsReplicatedWithExpirationTimes() {
+    public void queued_entries_with_expiration_times_are_not_lost_when_cluster_scaled_down() {
         final int entryCount = 1000;
-        final int ttlSec = 30;
+        final int ttlSec = 10;
+        final int writeDelaySec = 5;
+        final int backupCount = 1;
         final String mapName = randomMapName();
         TestEntryStore<Integer, Integer> testEntryStore = new TestEntryStore<>();
 
+        Config config = getConfigWithEntryStore(testEntryStore, writeDelaySec, backupCount);
+
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
-        HazelcastInstance[] instances = factory.newInstances(getConfigWithEntryStore(testEntryStore));
+        HazelcastInstance[] instances = factory.newInstances(config);
+
         IMap<Integer, Integer> map = instances[0].getMap(mapName);
         long expectedExpirationTime = System.currentTimeMillis() + ttlSec * 1000;
         for (int i = 0; i < 1000; i++) {
             map.put(i, i, ttlSec, TimeUnit.SECONDS);
         }
-        factory.terminate(instances[0]);
+
+        // scale down
+        instances[0].shutdown();
```

Comment:
Why don't we keep termination here? Since we wait for the safe state, the cluster should have already been balanced before the next instance's termination.

Reply:
Since by definition it is ungraceful, even we wait for safe state, it will be the safe state after ungraceful shutdown. I didn't see the benefit of terminate in the context of this test and to remove one unneeded complexity, replace it with shutdown.This was the reasoning.